### ISSUE TITLE
UMich Flux Hadoop is incorrectly detected as having a scheduler

### ISSUE THREAD

**Issue by bdice** [glotzerlab/signac-flow#60]
### Description

On host `flux-hadoop-login2.arc-ts.umich.edu`, signac-flow incorrectly detects the hostname pattern and thinks this is a cluster with a scheduler. I am not sure what the real problem is, though - I tried editing the hostname using the `suggested_pattern` below and still had an issue. I just know that the hostname is one likely part of the solution.

### To reproduce

```python
import flow.environments
import re
current_pattern = flow.environments.umich.FluxEnvironment().hostname_pattern  # '(nyx|flux).*.umich.edu'
suggested_pattern = '(nyx|flux)((?!hadoop).)*.umich.edu'
good_fqdn = 'flux-login2.arc-ts.umich.edu'
bad_fqdn = 'flux-hadoop-login2.arc-ts.umich.edu'
re.match(current_pattern, good_fqdn)  # returns a match on the torque cluster
re.match(current_pattern, bad_fqdn)  # returns a match on the hadoop cluster - this is wrong
re.match(suggested_pattern, good_fqdn)  # still returns a match on the torque cluster
re.match(suggested_pattern, bad_fqdn)  # does not return a match on the hadoop cluster - this is right
```

### Error output

```bash
bdice@flux-hadoop-login2 ~/project $ ./project.py status -d --show-traceback                                                 [5/1046]
Query scheduler...
WARNING:flow.project:Error occurred while querying scheduler: 'Torque not available.'.
ERROR:flow.project:Error occured during status update. Use '--show-traceback' to show the full traceback or '--ignore-errors' to complete the update anyways.
ERROR: Encountered error during program execution: 'Torque not available.'
Execute with '--show-traceback' or '--debug' to get more information.
Traceback (most recent call last):
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/scheduling/torque.py", line 29, in _fetch
    result = io.BytesIO(subprocess.check_output(cmd.split()))
  File "/sw/dsi/centos7/x86-64/Anaconda3-5.0.1/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/sw/dsi/centos7/x86-64/Anaconda3-5.0.1/lib/python3.6/subprocess.py", line 403, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/sw/dsi/centos7/x86-64/Anaconda3-5.0.1/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/sw/dsi/centos7/x86-64/Anaconda3-5.0.1/lib/python3.6/subprocess.py", line 1344, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'qstat': 'qstat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./project.py", line 86, in <module>
    Project().main()
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 2943, in main
    _exit_or_raise()
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 2911, in main
    args.func(args)
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 2571, in _main_status
    self.print_status(jobs=jobs, **args)
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 1242, in print_status
    self._fetch_scheduler_status(jobs, scheduler, err, ignore_errors)
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 1122, in _fetch_scheduler_status
    for sjob in self.scheduler_jobs(scheduler):
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 1000, in scheduler_jobs
    for sjob in self._expand_bundled_jobs(scheduler.jobs()):
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/project.py", line 976, in _expand_bundled_jobs
    for job in scheduler_jobs:
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/scheduling/torque.py", line 85, in jobs
    nodes = _fetch(user=self.user)
  File "/home/bdice/.local/lib/python3.6/site-packages/flow/scheduling/torque.py", line 31, in _fetch
    raise RuntimeError("Torque not available.")
RuntimeError: Torque not available.
```
### System configuration

 - Operating System [e.g. macOS]: Linux-3.10.0-862.el7.x86_64-x86_64-with-centos-7.5.1804-Core
 - Version of Python [e.g. 3.7]: 3.6.3 (Anaconda 64-bit, GCC 7.2.0)
 - Version of signac [e.g. 1.0]: 0.9.5
 - Version of signac-flow: 0.6.4

**Comment by bdice** [comment]
I used the pattern `'(nyx|flux)((?!-hadoop).)*.umich.edu'` instead of the one listed above (which is missing the hyphen). It corrected the problem I was seeing.

**Comment by bdice** [comment]
Resolved by PR #82.
